ocs_ci/utility/utils.py

```
@@ -1418,37 +1523,155 @@ def get_rosa_cli(
-                    f"Something went wrong with copying binary, reverting backup file. Exception: {ex}"
-                )
-                shutil.move(backup_file, binary_path)
-        else:
-            shutil.move(temp_binary, binary_path)
+
+        # Set executable permissions
+        current_file_permissions = os.stat(binary_path)
+        os.chmod(binary_path, current_file_permissions.st_mode | stat.S_IEXEC)
+
+    except Exception as e:
+        log.warning(f"Failed to extract {binary} from image {image}: {e}")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)